### ISSUE TITLE
Implement `backupFileExtension` as an option with `lib.homeManagerConfiguration` use-cases.

### ISSUE THREAD

**Issue by shymega** [nix-community/home-manager#5649]
### Description

I'm on NixOS 24.05 (plus HM 24.05), and I was wondering if it would be possible to allow the usage of `backupFileExtension` for Home Manager configurations using `lib.homeManagerConfiguration`. Currently, I can see the option is implemented when HM is used as a NixOS module, or a `nix-darwin` module. 

However, my config is based around the `lib.homeManagerConfiguration` module. I could embed it in the NixOS config, but on some systems, I can't install NixOS.

I've done basic searches in the issue tracker, and AFAICT, but this hasn't been implemented yet.

Thoughts?

**Comment by Liassica** [comment]
You could make `home-manager` an alias for `home-manager -b "<backup extension>"`; for example, I use `-b "hm.bak"`.

**Comment by Liassica** [comment]
If you use Home Manager to manage your shell, you can set aliases declaratively:
```nix
home.shellAliases.home-manager = "home-manager -b hm.bak";
```

**Comment by shymega** [comment]
This is true. However, I believe this should be implemented in the Nix `hmModule` itself.

When I get around to moving my shell config to Home Manager, I'll keep your suggestion in mind.

**Comment by shymega** [comment]
Looking through the exposed `homeManagerConfiguration` function via `lib`, I think it would be possible to use `backupFileExtension`, as an additional parameter.

I would like `homeManagerConfiguration` to have this argument and do not think using shell aliases solves the issue.

Would the maintainers be open to a PR adding this functionality?

Thank you.

**Comment by jmatsushita** [comment]
I would also benefit from this functionality for remote deployment of home manager profiles to ubuntu machines (using `deploy-rs`).

**Comment by s0racat** [comment]
+1 upvote for this feature

**Comment by astratagem** [comment]
Still an issue.

**Comment by mawkler** [comment]
I had to search the interwebs for half an hour before I found this issue because I thought I was just doing something wrong

**Comment by 0qln** [comment]
I have use cases for this aswell. Additionally, I think that the `backupCommand` option should be available for standalone builds aswell.
